tests/test_black.py

```diff
@@ -74,6 +74,50 @@ def read_data(name: str, data: bool = True) -> Tuple[str, str]:
     return "".join(_input).strip() + "\n", "".join(_output).strip() + "\n"
 
 
+@lru_cache()
+def list_fixtures(subdir: str) -> List[Path]:
+    base_dir = THIS_DIR / "data" / subdir
+    return [item for item in base_dir.iterdir() if item.suffix != ".out"]
+
+
+def python_source(request: Any) -> Tuple[str, str]:
+    item = cast(Path, request.param)
+    src = item.read_text(encoding="utf8")
+    outfile = item.with_suffix(".out")
+    if outfile.exists():
+        expected = outfile.read_text(encoding="utf8")
+    else:
+        if "# output\n" in src:
+            src, _, expected = (t.strip() + "\n" for t in src.partition("# output\n"))
+        else:
+            expected = src
+    return (src, expected)
+
+
+@pytest.fixture(
+    params=list_fixtures("python"),
+    ids=[str(item.relative_to(THIS_DIR)) for item in list_fixtures("python")],
```

Comment:
FWIW `ids` can take a function. Something like this `ids=lambda item: item.relative_to(THIS_DIR)` should work afair 🤞